Bottom: edbddd22f9dcc0f354bf616adaaac8e02ed0adac
Top:    ce1c63286e819e7c54ca6439ab908c5b95c3c93d
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2012-12-23 19:13:54 +0000

Remove PINFLAG, it is not required.

As nowhere cares that PINFLAG is set, we can avoid the code to handle this
and remove the flag.


---

diff --git a/src/const.h b/src/const.h
index 698eae4..0e858d1 100644
--- a/src/const.h
+++ b/src/const.h
@@ -116,8 +116,6 @@ values are reused for different object types.  The table below lists
 the numeric value followed by the symbolic name.
 
 @table @code
-@item 0x0001 pin
-If set, this object is a pin.  This flag is for internal use only.
 @item 0x0002 via
 Set on a rat if it should connect to a polygon with a via / pin and a thermal.
 @item 0x0004 found
@@ -182,7 +180,6 @@ Marker used internally to avoid revisiting an object.
 %end-doc */
 
 #define NOFLAG                  0x0000
-#define PINFLAG                 0x0001  /* is a pin */
 #define VIAFLAG                 0x0002  /* a rat should connect to a polygon with a via */
 #define FOUNDFLAG               0x0004  /* used by 'FindConnection()' */
 #define HOLEFLAG                0x0008  /* pin or via is only a hole */
diff --git a/src/create.c b/src/create.c
index 617fc4e..e1ab2a1 100644
--- a/src/create.c
+++ b/src/create.c
@@ -791,7 +791,6 @@ CreateNewPin (ElementType *Element,
   pin->Number = STRDUP (Number);
   pin->Flags = Flags;
   CLEAR_FLAG (WARNFLAG, pin);
-  SET_FLAG (PINFLAG, pin);
   pin->ID = ID++;
   pin->Element = Element;
 
diff --git a/src/strflags.c b/src/strflags.c
index d803e7e..e9a60f2 100644
--- a/src/strflags.c
+++ b/src/strflags.c
@@ -86,7 +86,6 @@ typedef struct
 } FlagBitsType;
 
 static FlagBitsType object_flagbits[] = {
-  { PINFLAG, N ("pin"), ALL_TYPES },
   { VIAFLAG, N ("via"), RATLINE_TYPES },
   { FOUNDFLAG, N ("found"), ALL_TYPES },
   { HOLEFLAG, N ("hole"), PIN_TYPES },
@@ -486,9 +485,6 @@ common_flags_to_string (FlagType flags,
     case RATLINE_TYPE:
       CLEAR_FLAG (RATFLAG, &fh);
       break;
-    case PIN_TYPE:
-      CLEAR_FLAG (PINFLAG, &fh);
-      break;
     }
 #endif
